dwitter/tests/api/test_api2_user.py

```diff
@@ -0,0 +1,190 @@
+                                      'user1Pass', 'user1Pass')
+
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(User.objects.count(), 0)
+
+    def test_email_long_but_valid(self):
+        # Django allows 254 character long emails
+        email = 'a'*64 + '@' + 'b' * 63 + '.com'
+        response = self.register_user('user1', email, 'user1Pass', 'user1Pass')
+        self.assertEqual(response.status_code, 201)
+        self.assertTrue(User.objects.filter(username='user1').exists())
+
+        newUser = User.objects.get(username='user1')
+        self.assertEqual(newUser.email, email)
+        self.assertTrue(newUser.check_password('user1Pass'))
+
+    def test_email_required(self):
+        response = self.register_user('user1', '', 'user1Pass', 'user1Pass')
+
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(User.objects.count(), 0)
+
+    def test_email_too_long(self):
+        response = self.register_user('user1', 'a'*256 + '@example.com', 'user1Pass', 'user1Pass')
+
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(User.objects.count(), 0)
+
+    def test_email_invalid(self):
+        response = self.register_user('user1', 'invalid_email', 'user1Pass', 'user1Pass')
+
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(User.objects.count(), 0)
+
+    def test_password_long_but_valid(self):
+        password = 'u1P!'*32  # 128 character long password
+        response = self.register_user('user1', 'user1@example.com', password, password)
+
+        self.assertEqual(response.status_code, 201)
+        self.assertTrue(User.objects.filter(username='user1').exists())
+
+        newUser = User.objects.get(username='user1')
+        self.assertEqual(newUser.email, 'user1@example.com')
+        self.assertTrue(newUser.check_password(password))
+
+    def test_password_required(self):
+        response = self.register_user('user1', 'user1@example.com', '', '')
+
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(User.objects.count(), 0)
+
+    def test_password_different(self):
+        response = self.register_user('user1', 'user1@example.com', 'user1Pass', 'differentPass2')
+
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(User.objects.count(), 0)
+
+    def test_password_bad(self):
+        response = self.register_user('user1', 'user1@example.com', 'user', 'user')
+
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(User.objects.count(), 0)
+
+
+class Api2UserGetTestCase(TransactionTestCase):
+    def setUp(self):
+        self.client = Client()
+        self.user1 = User.objects.create_user(
+            username='user1',
+            email='user1@example.com',
+            password='user1Pass',
+        )
+
+    def get_user(self, user_id):
+        return self.client.get(APIV2_PATH + 'users/' + str(user_id) + '/')
```

Comment:
Since you asked for an in-depth review: this can also be written as
```python
f'{APIV2_PATH}users/{user_id}/'
```

Reply:
Thanks, significantly easier to read ✅